jarust/src/jahandle.rs

```diff
@@ -105,11 +119,45 @@ impl JaHandle {
         self.send_request(request).await
     }
 
-    pub async fn message_with_jsep(&self, body: Value, jsep: Jsep) -> JaResult<JaResponse> {
+    /// Send a message and wait for the expected response
+    pub async fn message_with_result<Result>(&self, body: Value) -> JaResult<Result>
```

Comment:
```suggestion
    pub async fn message_with_result<R>(&self, body: Value) -> JaResult<R>
```

So we don't get confused with the `Result` type

Reply:
Also, I'm thinking of a better naming, as `message_with_result` here refers to send a message and wait on result. While `message_with_establishment_protocol` refers to send a message containing establishment protocol.

Reply:
Kept all the `_with` as they are for now